src/bin/julialauncher.rs

```diff
@@ -166,32 +166,49 @@ fn get_julia_path_from_channel(
     juliaupconfig_path: &Path,
     juliaup_channel_source: JuliaupChannelSource,
 ) -> Result<(PathBuf, Vec<String>)> {
+    let potential_matches: Vec<_> = config_data
+        .installed_channels
+        .keys()
+        .filter(|&item| {
+            item.starts_with(channel) && !item.starts_with("0") && !item.starts_with("1")
+        })
+        .cloned()
+        .collect();
+    let actual_channel = if potential_matches.len() == 1 {
+        potential_matches
+            .first()
+            .expect("length should be 1")
+            .as_str()
+    } else {
+        channel
+    };
+
     let channel_info = config_data
             .installed_channels
-            .get(channel)
+            .get(actual_channel)
```

Comment:
`channel`/`actual_channel` is not a good pair of variable names, imo.

`requested_channel`/`expanded_channel` could be a better choice.